storage/innobase/handler/handler0alter.cc

```diff
@@ -9921,6 +9810,35 @@ commit_try_rebuild(
 	}
 }
 
+/** Rename indexes in dictionary.
+@param[in]	ctx		alter info context
+@param[in]	ha_alter_info	Operation used during inplace alter
+@param[out]	trx		transaction to change the index name
+				in dictionary
+@return true if it failed to rename
+@return false if it is success. */
+static bool innobase_rename_index_try(
+	const ha_innobase_inplace_ctx*	ctx,
+	const Alter_inplace_info*	ha_alter_info,
+	trx_t*				trx)
+{
+	DBUG_ASSERT(ha_alter_info->handler_flags & ALTER_RENAME_INDEX);
+
+	for (const Alter_inplace_info::Rename_key_pair& pair :
+	     ha_alter_info->rename_keys) {
+		dict_index_t* index = dict_table_get_index_on_name(
+		    ctx->old_table, pair.old_key->name.str);
+		ut_ad(index);
+
+		if (innobase_rename_index_in_data_dictionary(
+			index, pair.new_key->name.str, trx)) {
+			return true;
+		}
```

Comment:
Again, what if `index` is not found? We shold avoid a crash, and return `true` (of course, after invoking `my_error()`).

The convention is that whenever we return `true`, we must have called `my_error()`.

Reply:
At this point `index != nullptr` because it was assured in `ha_innobase::prepare_inplace_alter_table()`.

`my_error_innodb()` is called inside `rename_index_try()`.